libs/shared/error/src/lib/error.ts

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
I spoke to Ben this morning and he had mentioned a desire to move these specific errors into each individual library, such that each library encapsulates it's own errors.

We'd then repurpose this file to instead export AccountError and PaymentError as reusable errors that can be extended. Advantages of this restructure:

1. We can check `instanceof` AccountError/PaymentError as an ancestor for all errors within the application, giving us insight into whether the error was instantiated by us.
2. We keep these library-specific errors within each library while still using the base errors from this library, which keeps errors close to the libraries that use them, so that this library doesn't bloat.

Reply:
This change could be made as a followup PR.

Reply:
@julianpoy : I assume this is part of a follow-up ticket to be filed and not blocking the current PR, which meets the acceptance criteria originally laid out for the ticket?

Reply:
See follow-up PR #15653 .

Reply:
Happy to create a follow-up ticket if needed for the gap between the current ticket and the new AC. Unsure if necessary given the follow-up PR, but happy to do so nonetheless given we need a reviewer.